llama_stack/distribution/routing_tables/toolgroups.py

```diff
@@ -52,9 +52,20 @@ async def list_tools(self, toolgroup_id: str | None = None) -> ListToolsResponse
 
         all_tools = []
         for toolgroup in toolgroups:
-            if toolgroup.identifier not in self.toolgroups_to_tools:
-                await self._index_tools(toolgroup)
-            all_tools.extend(self.toolgroups_to_tools[toolgroup.identifier])
+            try:
+                if toolgroup.identifier not in self.toolgroups_to_tools:
+                    await self._index_tools(toolgroup)
```

Comment:
I think the exception can only happen here and so the `try-catch` must only be around here.

Reply:
Good point - I narrowed this so only the _index_tools call is within the try block.